addons/binding/org.openhab.binding.seneye/src/main/java/org/openhab/binding/seneye/handler/SeneyeHandler.java

```diff
@@ -76,43 +76,31 @@ public void newState(SeneyeDeviceReading readings) {
 
     @Override
     public void invalidConfig() {
-        updateStatus(ThingStatus.OFFLINE);
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
     }
 
     @Override
     public void initialize() {
-        Configuration conf = this.getConfig();
+        SeneyeConfigurationParameters config = getConfigAs(SeneyeConfigurationParameters.class);
 
-        String param_aquariumname = String.valueOf(conf.get(PARAMETER_AQUARIUMNAME));
-        String param_username = String.valueOf(conf.get(PARAMETER_USERNAME));
-        String param_password = String.valueOf(conf.get(PARAMETER_PASSWORD));
-        String param_polltime = String.valueOf(conf.get(PARAMETER_POLLTIME));
-
-        if (param_aquariumname.isEmpty()) {
+        if (config.aquarium_name.isEmpty()) {

```

Comment:
are you sure that `config.aquarium_name` can never be `null` ?

Reply:
Added extra check, not for poll_time where a default value is set